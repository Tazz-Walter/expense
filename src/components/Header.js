import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/dashboard" exact={true} activeClassName="is-active">Dashboard</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></li>
            <li><NavLink to="/help" activeClassName="is-active">Help Page</NavLink></li>
            <li><button onClick={startLogout}>Log Out</button></li>
        </ul>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});
  
export default connect(undefined, mapDispatchToProps)(Header);