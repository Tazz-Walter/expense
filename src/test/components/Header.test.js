import React  from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';
import { shallow, mount, render } from 'enzyme';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});