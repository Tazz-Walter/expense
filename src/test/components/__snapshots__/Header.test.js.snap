// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    startLogout={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expensify
        </h1>,
        <ul>
          <li>
            <NavLink
              activeClassName="is-active"
              ariaCurrent="true"
              exact={true}
              to="/dashboard"
            >
              Dashboard
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="is-active"
              ariaCurrent="true"
              to="/create"
            >
              Create Expense
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="is-active"
              ariaCurrent="true"
              to="/help"
            >
              Help Page
            </NavLink>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              Log Out
            </button>
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expensify",
        },
        "ref": null,
        "rendered": "Expensify",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                exact={true}
                to="/dashboard"
              >
                Dashboard
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                to="/create"
              >
                Create Expense
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                to="/help"
              >
                Help Page
              </NavLink>
            </li>,
            <li>
              <button
                onClick={[Function]}
              >
                Log Out
              </button>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                exact={true}
                to="/dashboard"
              >
                Dashboard
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "ariaCurrent": "true",
                "children": "Dashboard",
                "exact": true,
                "to": "/dashboard",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                to="/create"
              >
                Create Expense
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "ariaCurrent": "true",
                "children": "Create Expense",
                "to": "/create",
              },
              "ref": null,
              "rendered": "Create Expense",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                to="/help"
              >
                Help Page
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "ariaCurrent": "true",
                "children": "Help Page",
                "to": "/help",
              },
              "ref": null,
              "rendered": "Help Page",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
              >
                Log Out
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log Out",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log Out",
              "type": "button",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expensify
          </h1>,
          <ul>
            <li>
              <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                exact={true}
                to="/dashboard"
              >
                Dashboard
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                to="/create"
              >
                Create Expense
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="is-active"
                ariaCurrent="true"
                to="/help"
              >
                Help Page
              </NavLink>
            </li>
            <li>
              <button
                onClick={[Function]}
              >
                Log Out
              </button>
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expensify",
          },
          "ref": null,
          "rendered": "Expensify",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                  activeClassName="is-active"
                  ariaCurrent="true"
                  exact={true}
                  to="/dashboard"
                >
                  Dashboard
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="is-active"
                  ariaCurrent="true"
                  to="/create"
                >
                  Create Expense
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="is-active"
                  ariaCurrent="true"
                  to="/help"
                >
                  Help Page
                </NavLink>
              </li>,
              <li>
                <button
                  onClick={[Function]}
                >
                  Log Out
                </button>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="is-active"
                  ariaCurrent="true"
                  exact={true}
                  to="/dashboard"
                >
                  Dashboard
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "ariaCurrent": "true",
                  "children": "Dashboard",
                  "exact": true,
                  "to": "/dashboard",
                },
                "ref": null,
                "rendered": "Dashboard",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="is-active"
                  ariaCurrent="true"
                  to="/create"
                >
                  Create Expense
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "ariaCurrent": "true",
                  "children": "Create Expense",
                  "to": "/create",
                },
                "ref": null,
                "rendered": "Create Expense",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="is-active"
                  ariaCurrent="true"
                  to="/help"
                >
                  Help Page
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "ariaCurrent": "true",
                  "children": "Help Page",
                  "to": "/help",
                },
                "ref": null,
                "rendered": "Help Page",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
                >
                  Log Out
                </button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log Out",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Log Out",
                "type": "button",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
